packages/flutter_tools/test/general.shard/base/analyze_size_test.dart

```diff
@@ -0,0 +1,163 @@
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/analyze_size.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/process.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+void main() {
+  MemoryFileSystem fileSystem;
+  BufferLogger logger;
+  FakeProcessManager processManager;
+
+  const FakeCommand unzipCommmand = FakeCommand(
+    command: <String>[
+      'unzip',
+      '-o',
+      '-v',
+      'test.apk',
+      '-d',
+      '/.tmp_rand0/flutter_tools.rand0'
+    ],
+    stdout: '''
+Length   Method    Size  Cmpr    Date    Time   CRC-32   Name
+--------  ------  ------- ---- ---------- ----- --------  ----
+11708  Defl:N     2592  78% 00-00-1980 00:00 07733eef  AndroidManifest.xml
+1399  Defl:N     1092  22% 00-00-1980 00:00 f53d952a  META-INF/CERT.RSA
+46298  Defl:N    14530  69% 00-00-1980 00:00 17df02b8  META-INF/CERT.SF
+46298  Defl:N    14530  69% 00-00-1980 00:00 17df02b8  lib/arm64-v8a/libapp.so
+46298  Defl:N    14530  69% 00-00-1980 00:00 17df02b8  lib/arm64-v8a/libflutter.so
+''',
+  );
+
+  const String aotSizeOutput = '''
+[
+    {
+        "l": "dart:_internal",
+        "c": "SubListIterable",
+        "n": "[Optimized] skip",
+        "s": 2400
+    },
+    {
+        "l": "dart:_internal",
+        "c": "SubListIterable",
+        "n": "[Optimized] new SubListIterable.",
+        "s": 3560
+    },
+    {
+        "l": "dart:core",
+        "c": "RangeError",
+        "n": "[Optimized] new RangeError.range",
+        "s": 3920
+    },
+    {
+        "l": "dart:core",
+        "c": "ArgumentError",
+        "n": "[Stub] Allocate ArgumentError",
+        "s": 4650
+    }
+]
+''';
+
+  setUp(() {
+    fileSystem = MemoryFileSystem.test();
```

Comment:
ah is this why those parameters were in the constructor? There is already a dependency injection mechanism. See other tests using `testUsingContext`. Then you can remove those construction params.

Reply:
Yes, new code should not use the global zone injection - it makes testing the code very brittle as we've found that new contributors frequently do not understand which interfaces need to be overriden.

Reply:
So the new consensus is to do constructor injection through the entire graph? Is there a tracking bug to migrate existing code?

Reply:
https://github.com/flutter/flutter/issues/47161 , but I haven't done a good job of keeping it updated